### PR TITLE
DNS Audit Log

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Log all the dns request from pfdns.

# Impacts
DNS

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* DNS Audit Log

### CODE REVIEW

**Comment by extrafu** [comment]
Please rebase.

**Comment by extrafu** [comment]
@jrouzierinverse Rewiew it one last time.

@lzammit Test it.

File: lib/pf/dal/_dns_audit_log.pm [hunk 1]
```diff
@@ -0,0 +1,268 @@
+package pf::dal::_dns_audit_log;
```
Comment (jrouzierinverse) [inline lib/pf/dal/_dns_audit_log.pm:1]: Rerun generator-data-access-layer

```bash
perl addons/dev-helpers/bin/generator-data-access-layer.pl
```

**Comment by fdurand** [comment]
Bump

**Comment by julsemaan** [comment]
This has the potential of creating an enormous amount of DNS queries. Can this be fully disabled in pfdns using configuration? Otherwise we expose ourselves to an enormous amount of writes on the disk with this feature

**Pull request merged**